src/Console/SelfUpdate/NewVersionChecker.php

```diff
@@ -19,7 +19,7 @@
 /**
  * @internal
  */
-final class NewVersionChecker
+class NewVersionChecker
```

Comment:
this is done for `prophesize` onyl?

Reply:
Yes. This is exactly the reason of my comment on [#3183](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3183#discussion_r147670555).

Reply:
please bring final and create interface.

Reply:
TBH I don't see the point in creating an interface for testing purposes only.

Reply:
`final` prevents partial mocks, which are extra bad approach imo.
i like to be defensive.

Reply:
What do you mean partial mocks? Mocks with some methods defined in the test and some not defined? If so, using an interface does not prevent this. Also, when using Prophecy, as soon as a mock has at least one defined method, all methods used in the tested code must be defined as well. Calling an undefined method will throw an exception, which should prevent the issue.

Reply:
with built-in mock system of PHPUnit, it's possible to mock real class with using mocked one method but not another. final class, will forbid that totally.

Reply:
We don't use PHPUnit mocks but Prophecy anyway so this still looks pointless to me.

Reply:
final or not is a general rule. using final is being defensive. partial mocks were example, not the only reason.